### PR TITLE
Add a section guiding through the process of rule divergence

### PR DESCRIPTION
**Pull request by matejak**
#### Description:

Expand the dev guide with content regarding how to deal with changes in systems that could be typically solved either by complicating existing rules, or creating new ones. The new section makes it easier to pick the right poison.

#### Rationale:

Although that class of problems typically doesn't have clearly good or clearly bad solutions, approaching the problem systematically is always beneficial.

### CODE REVIEW

**Comment by Mab879** [comment]
The failures on Fedora Rawhide is expected due to OpenSCAP/openscap#1995.

File: docs/manual/developer/03_creating_content.md [hunk 1]
```diff
@@ -1301,3 +1301,229 @@ Each rule in the benchmark in the `/linux_os/guide` directory must be a member o
 Products specify a path to the directory with component files by the `components_root` key in the `product.yml`.
 
 The mappings of components to rules can be converted to HTML by the `utils/render_components.py` script.
+
+
+## Preventing problems
```
Comment (Mab879) [inline docs/manual/developer/03_creating_content.md:1306]:
```suggestion
## Preventing Problems
```

File: docs/manual/developer/03_creating_content.md [hunk 2]
```diff
@@ -1301,3 +1301,229 @@ Each rule in the benchmark in the `/linux_os/guide` directory must be a member o
...
+- The original rule and the spawned one are clear and direct about what they do.
+- The increased rule granularity promotes reusability, making such rules great  blocks for profiles and controls.
+
+Cons
+
+- Too much granularity in profiles and controls may create noise and impact the user experience.
+- Spawning may result in creation of almost identical rules without any real benefit of granularity.
+- Bugs affecting multiple spawned rules may be more laborious to be consistently fixed along the time.
+
+Profile and control selections
```
Comment (Mab879) [inline docs/manual/developer/03_creating_content.md:1430]: Seems like something is missing here

File: docs/manual/developer/03_creating_content.md [hunk 3]
```diff
@@ -1301,3 +1301,229 @@ Each rule in the benchmark in the `/linux_os/guide` directory must be a member o
...
+
+##### Conditionals
+
+Pros
+
+- Unlike with spawning, no changes in profile selections, controls or tailoring.
+
+
```
Comment (Mab879) [inline docs/manual/developer/03_creating_content.md:1438]:
```suggestion
```

File: docs/manual/developer/03_creating_content.md [hunk 4]
```diff
@@ -1301,3 +1301,229 @@ Each rule in the benchmark in the `/linux_os/guide` directory must be a member o
...
+##### Spawning
+
+Remarks
+
+- The ability to define extendable controls may mitigate the shortcomings mentioned right above.
+
+Content Clarity
```
Comment (Mab879) [inline docs/manual/developer/03_creating_content.md:1445]: I would suggest making these pseudo headings bold, to help communicate these are heading like objects.

File: docs/manual/developer/03_creating_content.md [hunk 5]
```diff
@@ -1301,3 +1301,229 @@ Each rule in the benchmark in the `/linux_os/guide` directory must be a member o
...
+
+- a component changed between minor product versions, or
+- behaves differently on different architectures.
+
+These divergences emerge as the result of continued support of a minor version, in other words, once a new minor version of a product is released the previous version doesn’t go End Of Life immediately and needs support.
+
+Examples of in-product divergences are:
+
+- Configuration of [SSHD Compression](https://github.com/ComplianceAsCode/content/blob/328eac5d78ee756d158c389a91633f5dd74a5d60/linux_os/guide/services/ssh/ssh_server/sshd_disable_compression/rule.yml#L52), which makes sense only on rhel < 7.4
+- Configuration of [systemd's StopIdleSessionSec](https://github.com/ComplianceAsCode/content/blob/328eac5d78ee756d158c389a91633f5dd74a5d60/linux_os/guide/system/accounts/accounts-physical/logind_session_timeout/rule.yml#L22), which are available on rhel >= 8.7 and rhel > 9.0
```
Comment (Mab879) [inline docs/manual/developer/03_creating_content.md:1362]:
```suggestion
- Configuration of [systemd's StopIdleSessionSec](https://github.com/ComplianceAsCode/content/blob/328eac5d78ee756d158c389a91633f5dd74a5d60/linux_os/guide/system/accounts/accounts-physical/logind_session_timeout/rule.yml#L22), which are available on rhel >= 8.7 and rhel > 9.0.
```

**Review by Mab879 — CHANGES_REQUESTED** [review]
Thanks for the PR I have left many comments through the file.

I will note that I had hard time figuring out how the headings where working under "Aspects to consider when picking one approach" section. I would suggest at least using bold for text that you are using a header. 


For example:
```Stability

##### Conditionals
```

If I'm reading the markdown file I think that is something missing (it's not as bad when viewing the rendered file, but screen readers that depend section headers might get confused. I  was assuming header was starting something new. However, I believe that you using `Stability` as some sort of header that is above (logically) Conditionals.

If you have any other questions please reach out.

File: docs/manual/developer/03_creating_content.md [hunk 6]
```diff
@@ -1301,3 +1301,229 @@ Each rule in the benchmark in the `/linux_os/guide` directory must be a member o
...
+Pros
+
+- Opportunity to remove “dead” code sooner than a “dead” rule, since a rule cannot be removed from a product's data stream, but unnecessary code can.
+
+Cons
+
+- Getting rid of “dead” conditionals requires refactoring.
+- Unless actively pruned, the conditionals in the rule might stay there indefinitely.
+
+##### Spawning
```
[the remark this answers is not in the document]
Comment (matejak) [reply docs/manual/developer/03_creating_content.md:1525]: no cons from this POV.

File: docs/manual/developer/03_creating_content.md (hunk 2 above)
Comment (matejak) [reply docs/manual/developer/03_creating_content.md]: Headings should be there.

**Comment by Mab879** [comment]
Waving the Automatus Sanity, the fail is not related to this PR.

**Pull request merged**